russstation/code/modules/hydroponics/plant_genes.dm

```diff
@@ -0,0 +1,34 @@
+
+/**
+ * Spawns a confetti gib when someone slips on it.
+ * Because it's funny :)
+ * Just like slippery skin, if we have a trash type this only functions on that. (Banana peels)
+ */
+/datum/plant_gene/trait/plant_confetti
+	name = "Farceur Emission"
+	mutability_flags = PLANT_GENE_REMOVABLE | PLANT_GENE_MUTATABLE | PLANT_GENE_GRAFTABLE
+
+/datum/plant_gene/trait/plant_confetti/on_new_plant(obj/item/our_plant, newloc)
+	. = ..()
+	if(!.)
+		return
+
+	var/obj/item/food/grown/grown_plant = our_plant
+	if(istype(grown_plant) && ispath(grown_plant.trash_type, /obj/item/grown))
+		return
+
+	RegisterSignal(our_plant, COMSIG_PLANT_ON_SLIP, .proc/confetti)
```

Comment:
does this need a matching unregister signal?

Reply:
Not that I'm aware of; this is more or less a copy/paste from the existing `code\modules\hydroponics\plant_genes.dm` file from TG Upstream. 

Searching through the code I can't find mention of unregistering anything for `COMSIG_PLANT_ON_SLIP` unless I'm missing it or it's under a different reference?